src/Orleans/Streams/PersistentStreams/StreamQueueBalancerType.cs

```diff
@@ -39,5 +39,10 @@ public enum StreamQueueBalancerType
         /// That is, it does not rebalance queues based on dymanic changes in the cluster Membership.
         /// </summary>
         StaticClusterConfigDeploymentBalancer,
+
+        /// <summary>
+        /// Use user configured custom queue balancer
+        /// </summary>
+        CustomBalancer
```

Comment:
As to add balancers to DI. I suspect the value of it. Objects are put into DI because it needs to be available in multiple places. But for balancers, it got created once per stream provider and pass to pulling agent manager, and it's done. I don't see where DI comes to play here. 

And adding balancer to DI doesn't play well with its life cycle. every stream provider create balancer at the step of initializing pulling agent manager, with a different set of params per stream provider. This is hard to access and differentiate in the step of building DI container 

It make sense to me to add BalancerFactory to DI as a named service, because that's the most convenient and familiar pattern for Orleans user to configure a custom service now.